agent/eni/networkutils/utils_windows.go

```diff
@@ -106,12 +109,59 @@ func (utils *networkUtils) retrieveMAC() (string, error) {
 	return utils.macAddress, nil
 }
 
-// Returns all the network interfaces
+// GetAllNetworkInterfaces returns all the network interfaces.
 func (utils *networkUtils) GetAllNetworkInterfaces() ([]net.Interface, error) {
 	return utils.netWrapper.GetAllNetworkInterfaces()
 }
 
-// This method is used to inject netWrapper instance. This will be handy while testing to inject mocks.
+// SetNetWrapper is used to inject netWrapper instance. This will be handy while testing to inject mocks.
 func (utils *networkUtils) SetNetWrapper(netWrapper netwrapper.NetWrapper) {
 	utils.netWrapper = netWrapper
 }
+
+// GetDNSServerAddressList returns the DNS server addresses of the queried interface.
+func (utils *networkUtils) GetDNSServerAddressList(macAddress string) ([]string, error) {
+	addresses, err := utils.netWrapper.GetAdapterAddresses()
+	if err != nil {
+		return nil, err
+	}
+
+	var firstDnsNode *windows.IpAdapterDnsServerAdapter
+	// Find the adapter which has the same mac as queried.
+	for _, adapterAddr := range addresses {
```

Comment:
Why not append the DNS server addresses to the return list directly?

Reply:
The DNS server list returned from Win32 API call is a raw linked list implementation whereas the data type expected by the caller is a Golang slice of strings.
We need to iterate each node of the list and parse the SocketAddress into a IP address and add them to the slice.